test/Dialect/ESI/windows.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: circt-opt %s -verify-diagnostics | circt-opt -verify-diagnostics | FileCheck %s
+
+!typeA = !hw.struct<header1: i6, header2: i1, header3: i16>
+
+// CHECK-LABEL:   esi.window @TypeAwin1 into !hw.struct<header1: i6, header2: i1, header3: i16> {
+// CHECK-NEXT:      esi.window.frame {
+// CHECK-NEXT:        esi.window.field "header1" : i6
+// CHECK-NEXT:        esi.window.field "header2" : i1
+// CHECK-NEXT:      }
+// CHECK-NEXT:      esi.window.frame {
+// CHECK-NEXT:        esi.window.field "header3" : i16
+// CHECK-NEXT:      }
+// CHECK-NEXT:    }
+esi.window @TypeAwin1 into !typeA {
+  esi.window.frame {
+    esi.window.field "header1" : i6
+    esi.window.field "header2" : i1
+  }
+  esi.window.frame {
+    esi.window.field "header3" : i16
+  }
+}
+
+
+// CHECK-LABEL:   hw.module.extern @TypeAModuleDst(%windowed: !esi.window<@typeAwin1>)
+hw.module.extern @TypeAModuleDst(%windowed: !esi.window<@typeAwin1>) 
```

Comment:
I think `typeAwin1` should be `TypeAwin1`, and the fact that this isn't caught points to the disadvantages you mention :)

Reply:
Yep. I noticed that before I opened this PR but kept it in there to make that point. Good eyes!